client/my-sites/stats/stats-notices/index.tsx

```diff
@@ -67,18 +69,30 @@ const NewStatsNotices = ( { siteId, isOdysseyStats }: StatsNoticesProps ) => {
 	);
 };
 
-const PostPurchaseNotices = ( { siteId, statsPurchaseSuccess }: StatsNoticesProps ) => {
+const PostPurchaseNotices = ( {
+	siteId,
+	statsPurchaseSuccess,
+	isOdysseyStats,
+}: StatsNoticesProps ) => {
 	// Check if the GET param is passed to show the Free or Paid plan purchase notices
 	const showFreePlanPurchaseSuccessNotice = statsPurchaseSuccess === 'free';
 	const showPaidPlanPurchaseSuccessNotice = statsPurchaseSuccess === 'paid';
 
+	const [ paramRemoved, setParamRemoved ] = useState( false );
+
 	const removeParam = () => {
-		if ( ! statsPurchaseSuccess ) {
+		if ( ! statsPurchaseSuccess || paramRemoved ) {
 			return;
 		}
-		const newUrl = removeStatsPurchaseSuccessParam( window.location.href );
+		// Ensure it runs only once.
+		setParamRemoved( true );
+		const newUrlObj = removeStatsPurchaseSuccessParam( window.location.href, !! isOdysseyStats );
+		if ( isOdysseyStats ) {
+			// We need to update the page base if it changed. Otherwise, pagejs won't be able to find the routes.
+			page.base( `${ newUrlObj.pathname }${ newUrlObj.search }` );
+		}
 		// Odyssey would try to hack the URL on load to remove duplicate params. We need to wait for that to finish.
-		setTimeout( () => window.history.replaceState( null, '', newUrl ), 300 );
+		setTimeout( () => window.history.replaceState( null, '', newUrlObj.toString() ), 300 );
```

Comment:
Would it make sense to move `page.base( ... )` invocation into the setTimeout callback?

Reply:
Yes it does make sense. It would be safer doing that. Will do a follow up PR since the PR is merged already. Thanks for the suggestion.